src/backend/src/guards.rs

```diff
@@ -8,3 +9,20 @@ pub fn caller_is_not_anonymous() -> Result<(), String> {
         Ok(())
     }
 }
+
+pub fn caller_is_allowed() -> Result<(), String> {
+    let caller = caller();
+    let allowed_callers = read_state(|s| s.allowed_callers.clone());
+
+    fn is_allowed_callers(caller: Principal, allowed_callers: &Vec<Principal>) -> bool {
+        allowed_callers
+            .iter()
+            .any(|allowed_caller| *allowed_caller == caller)
+    }
+
+    if is_allowed_callers(caller, &allowed_callers) {
```

Comment:
I'd do it a bit simpler:

```suggestion
    if read_state(|s| s.allowed_callers.contains(&caller())) {
```